pkg/dependencies/factory.go

```diff
@@ -1046,8 +1046,12 @@ func (f *Factory) WithVSphereValidator() *Factory {
 		if f.dependencies.VSphereValidator != nil {
 			return nil
 		}
-
-		f.dependencies.VSphereValidator = vsphere.NewValidator(f.dependencies.Govc, &networkutils.DefaultNetClient{})
+		vb := vsphere.NewValidatorBuilder(
```

Comment:
Take a look at what we adopted now, we decided not to use a validator builder but instead pull the clients as part of the validator methods: https://github.com/aws/eks-anywhere/pull/2887